src/targets/services/konnectorAlerts.js

```diff
@@ -1,312 +1,11 @@
-    Q(TRIGGER_DOCTYPE).where({
-      worker: 'konnector'
-    })
-  )
-  const triggerStatesDoc = await fetchTriggerStates(client)
-  const previousStates = get(triggerStatesDoc, 'triggerStates', {})
-  logger('info', `${cronKonnectorTriggers.length} konnector triggers`)
-
-  const ignoredErrorFlag = flag('banks.konnector-alerts.ignored-errors')
-  const ignoredErrors = new Set(
-    ignoredErrorFlag ? ignoredErrorFlag.split(',') : []
-  )
-
-  const triggerAndNotifsInfo = (await Promise.all(
-    cronKonnectorTriggers.map(async trigger => {
-      return {
-        trigger,
-        shouldNotify: await shouldNotify({
-          client,
-          trigger,
-          previousStates,
-          serviceTrigger
-        })
-      }
-    })
-  )).filter(({ trigger, shouldNotify }) => {
-    if (shouldNotify.ok || ignoredErrors.has(shouldNotify.reason)) {
-      logger('info', `Will notify trigger for ${getKonnectorSlug(trigger)}`)
-      return true
-    } else {
-      logger(
-        'info',
-        `Will not notify trigger for ${getKonnectorSlug(trigger)} because ${
-          shouldNotify.reason
-        }`
-      )
-      return false
-    }
-  })
-
-  const konnectorSlugs = uniq(
-    triggerAndNotifsInfo.map(({ trigger }) => getKonnectorSlug(trigger))
-  )
-
-  const konnectorNamesBySlug = fromPairs(
-    await Promise.all(
-      konnectorSlugs.map(async slug => [
-        slug,
-        get(
-          await fetchRegistryInfo(client, slug),
-          'latest_version.manifest.name',
-          slug
-        )
-      ])
-    )
-  )
-
-  logger('info', `${triggerAndNotifsInfo.length} konnector triggers to notify`)
-
-  if (triggerAndNotifsInfo.length) {
-    const notifView = buildNotification(client, {
-      konnectorAlerts: triggerAndNotifsInfo.map(({ trigger }) => {
-        const konnectorSlug = trigger.message.konnector
-        return {
-          konnectorSlug,
-          konnectorName: konnectorNamesBySlug[konnectorSlug] || konnectorSlug,
-          trigger
-        }
-      })
-    })
-    await sendNotification(client, notifView)
-  }
-
-  await storeTriggerStates(client, cronKonnectorTriggers, triggerStatesDoc)
-}
-
-export const destroyObsoleteTrigger = async (client, trigger) => {
-  if (trigger?.type === '@at') {
-    logger('info', 'Try to destroy @at trigger...')
-
-    const isObsolete = +new Date(trigger?.arguments) < +new Date()
-
-    if (isObsolete) {
-      await client.destroy(trigger)
-      logger('info', `Destroyed @at trigger with id ${trigger._id}`)
-    } else {
-      logger(
-        'info',
-        `Nothing happened, trigger with id ${trigger._id} is not yet obsolete`
-      )
-    }
-  }
-}
+import { TRIGGER_DOCTYPE, JOBS_DOCTYPE } from 'doctypes'
+import { logger } from 'ducks/konnectorAlerts'
+import { runService } from './service'
+import { destroyObsoleteTrigger } from './konnectorAlerts/helpers'
+import { sendTriggerNotifications } from './konnectorAlerts/sendTriggerNotifications'
```

Comment:
For this project, you should avoir the use of relative import. You should use aliases all the time. It's easier to override

Reply:
J'ai vu ça mais je n'ai pas réussi à l'implémenter rapidement. C'est le `npm run build:services` qui déconne, l'import ne marchait pas.

J'ai testé en rajoutant un `services: './src/targets/services'` dans https://github.com/cozy/cozy-banks/blob/feat/notifications/babel.config.js#L28et un `services: path.resolve(__dirname, 'src/targets/services/')` dans https://github.com/cozy/cozy-banks/blob/feat/notifications/webpack.madge.js#L10

Je vois qu'on a un `appServicesFolder` dans cozy-scripts https://github.com/cozy/create-cozy-app/blob/master/packages/cozy-scripts/utils/paths.js#L77, on peut peut-être rajouter un alias dans la lib plutôt que dans l'app banks ?